taxi/tests/test_views.py

```diff
@@ -0,0 +1,396 @@
+
+
+class PrivetCarTest(TestCase):
+    def setUp(self) -> None:
+        self.driver = get_user_model().objects.create_user(
+            username="driver",
+            password="driver55"
+        )
+        self.manufacturer = Manufacturer.objects.create(
+            name="test",
+            country="country"
+        )
+        self.client.force_login(self.driver)
+
+    def test_retrieve_car_list(self):
+        car = Car.objects.create(
+            model="test_model",
+            manufacturer=self.manufacturer
+        )
+        car.drivers.add(self.driver)
+        car_1 = Car.objects.create(
+            model="model_1",
+            manufacturer=self.manufacturer
+        )
+        car_1.drivers.add(self.driver)
+        res = self.client.get(CAR_URL)
+        self.assertEqual(res.status_code, 200)
+        cars = Car.objects.all()
+        self.assertEqual(
+            list(res.context["car_list"]),
+            list(cars)
+        )
+        self.assertTemplateUsed(res, "taxi/car_list.html")
+
+    def test_create_car(self):
+        response = self.client.post(reverse("taxi:car-create"), {
+            "model": "New Car",
+            "manufacturer": self.manufacturer.id,
+            "drivers": self.driver.id
+        })
+        self.assertEqual(response.status_code, 302)
+        self.assertTrue(Car.objects.filter(model="New Car").exists())
+
+    def test_update_car(self):
+        car = Car.objects.create(
+            model="Car",
+            manufacturer=self.manufacturer,
+        )
+        car.drivers.set([self.driver.id])
+        form_data = {
+            "model": "new car",
+            "manufacturer": self.manufacturer.id,
+            "drivers": self.driver.id
+        }
+        resp = self.client.post(
+            reverse("taxi:car-update", args=[car.id]),
+            form_data
+        )
+
+        self.assertEqual(resp.status_code, 302)
+        car.refresh_from_db()
+        self.assertEqual(car.model, "new car")
+
+    def test_delete_car(self):
+        car = Car.objects.create(
+            model="Car to Delete",
+            manufacturer=self.manufacturer
+        )
+        response = self.client.post(reverse("taxi:car-delete", args=[car.id]))
+        self.assertEqual(response.status_code, 302)
+        self.assertFalse(Car.objects.filter(id=car.id).exists())
+
+
+class CarSearchFormTest(TestCase):
+    def setUp(self):
+        self.manufacturer = Manufacturer.objects.create(
+            name="test",
+            country="country"
+        )
+        self.car_1 = Car.objects.create(
+            model="Car 1",
+            manufacturer=self.manufacturer
+        )
+        self.car_2 = Car.objects.create(
+            model="Car 2",
+            manufacturer=self.manufacturer
+        )
+        self.car_3 = Car.objects.create(
+            model="Car 3",
+            manufacturer=self.manufacturer
+        )
+
+    def test_search_car_with_valid_data(self):
```

Comment:
Would make sense to add a case when you search for "Car 1" and then check if "Car 1" is in results and "Car 2 & 3" is not in results at the same time